core/services/ocr2/plugins/ocr2keeper/evm21/transmit_event_provider.go

```diff
@@ -211,12 +211,21 @@ func (c *TransmitEventProvider) convertToTransmitEvents(logs []transmitEventLog,
 	for _, l := range logs {
 		upkeepId := ocr2keepers.UpkeepIdentifier(l.Id().Bytes())
 		triggerID := l.TriggerID()
+		//TODO: is this the right way to get the trigger bytes?
+		//Unpack the trigger, triggerID is byte[32]
+		//triggerW, _ := UnpackTrigger(l.Id(), triggerID[:])
+
+		// TODO: need to manually convert trigger wrapper to trigger
```

Comment:
As said in TODOs, first need to confirm how to construct Trigger object from a log. Second is to convert Trigger Wrapper to Trigger object which we can use to generate workID.